### PR TITLE
Fix the spider_base template

### PR DESCRIPTION
**Pull request by octopusinvitro**
The `spider_base` template now uses the `countries` method instead of manipulating a JSON object.

### CODE REVIEW

File: views/all_countries.erb [hunk 1]
```diff
@@ -1,11 +1,13 @@
 <div class="page-section" id="home">
     <div class="container">
         <ul class="grid-list grid-list--vertically-center">
-          <% @page.world.sort_by { |k, _| k }.each do |k, v| %>
+          <% @page.world.as_json.sort_by { |key, _| key }.each do |key, value| %>
```
Comment (tmtmtmtm) [inline views/all_countries.erb]: The goal here is to kill that `as_json` method. You'll probably want to pass `world.all_countries` or equivalent into the template which provides a list of country objects.

File: lib/world.rb [hunk 2]
```diff
@@ -10,6 +10,10 @@ def as_json
     @_wjson ||= JSON.parse(File.read(file), symbolize_names: true)
   end
 
+  def countries
+    as_json.map { |slug, _| country(slug) }
```
Comment (tmtmtmtm) [inline lib/world.rb]: Would probably be a little clearer to do that as `as_json.keys.map { |slug| country_slug }`

Comment (tmtmtmtm) [inline lib/world.rb]: I'd probably also have this guarantee to return the countries already sorted (by name). Most uses are probably going to want that, so may as well save them the work.

File: lib/page/spider_base.rb [hunk 3]
```diff
@@ -3,12 +3,18 @@
...
-    def world
-      World.new.as_json
-    end
-
     def title
       'EveryPolitician: Robots Start Here'
     end
+
+    def countries_sorted
+      world.countries.sort_by(&:slug)
```
Comment (tmtmtmtm) [inline lib/page/spider_base.rb]: Sorting by slug is the existing behaviour, but as this is essentially a private page, I don't think it's essential to maintain that order. So if `world.countries` were to return a list ordered by name, I think it would be OK to pass it straight through without resorting here.

File: t/web/spider_base.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+require 'test_helper'
+require_relative '../../app'
+require 'pry'
```
Comment (tmtmtmtm) [inline t/web/spider_base.rb]: I don't think you want to leave that behind. (Though might be worth making a separate PR to just add it to `test_helper` so that all tests have it available)

**Comment by octopusinvitro** [comment]
@tony I rebased the branchso tidy things up a bit.

**Comment by tmtmtmtm** [comment]
LGTM!
